ksql-version-metrics-client/src/main/java/io/confluent/ksql/version/metrics/collector/BasicCollector.java

```diff
@@ -26,10 +27,18 @@
 
   private final TimeUtils timeUtils;
   private final KsqlModuleType moduleType;
+  private final ActiveChecker activeChecker;
 
-  public BasicCollector(final KsqlModuleType moduleType, final TimeUtils timeUtils) {
+  // 24 hours
+  private static final int maxInterval = 24 * 60 * 60 * 1000;
+
+  public BasicCollector(
+      final KsqlModuleType moduleType,
+      final TimeUtils timeUtils,
+      final ActiveChecker activeChecker) {
```

Comment:
In a similar way that `KsqlResource` does not need an instance of `ActiveChecker`, neither does `BasicCollector`. Just pass `Supplier<Boolean> activeSupplier`, as that is _all_ this class needs.  This will also help simplify and decouple your tests, i.e. you no longer need an instance of `ActiveChecker` in your test for this class.


```suggestion
      final Supplier<Boolean> activeSupplier) {
```